src/components/TextInput/TextInput.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import colors from '../../theme/colors.scss';
+
+import './TextInput.scss';
+
+const prefixCls = 'kai-text-input';
+
+class TextInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFocused: false,
+    };
+    this.handleFocusChange = this.handleFocusChange.bind(this);
+  }
+
+  handleFocusChange(isFocused) {
+    this.setState({ isFocused });
+    if (isFocused) {
+      this.props.onFocusChange(this.props.index);
+      this.props.forwardedRef.current.focus();
+    }
+  }
+
+  render() {
+    const { focusColor, forwardedRef, label, onChange } = this.props;
+    const { isFocused } = this.state;
+
+    const itemCls = classnames([
+      prefixCls,
+      isFocused && `${prefixCls}--focused`,
+    ]);
+    const labelCls = `${prefixCls}-label`;
+    const inputCls = `${prefixCls}-input`;
+
+    return (
+      <div
+        tabIndex="0"
+        className={itemCls}
+        style={{ backgroundColor: isFocused ? focusColor : colors.white }}
+        onFocus={() => this.handleFocusChange(true)}
+        onBlur={() => this.handleFocusChange(false)}
+      >
+        <label className={labelCls}>{label}</label>
+        <input
```

Comment:
One bug I noticed is that when this element resides within a `TabView` hitting left or right moves the cursor as well as switches tabs. Currently there seems to be no way of actually moving the cursor in this `TextInput`, however we might be able to take advantage of the softkeys here to do so. Alternatively we can try to override the `TabView` behaviour but that doesn't seem like good design. What do you think?

Reply:
Hmmm... This is a tricky one. I can't seem to find any prior art from poking around my phone, which perhaps suggests an input in a tab is just a bad idea. We could use the soft key to allow the user to "exit" from the input, but all options feel a bit janky.

I wonder is it best to just not support input inside a tab? Or implement something simple, but janky and document it to basically tell users they probably shouldn't put an input in a tab?

Reply:
Not supporting it could definitely work however I don't want to limit use cases yet without some investigation. Maybe try using the softkeys for this (ex. register left softkey to move left by one character) and see what the experience is like?

Reply:
Yeah, I'll give that a go. I'd say it's a decent workaround, but should probably be avoided by anyone using the lib if they have other ways of structuring their app.

Reply:
Yup, just a quick experiment might be cool, but I agree with you in principle

Reply:
Had a thought on this. Instead of messing around with the soft keys, we could make it that it only swaps tabs if you hit left/right at the start/end of the input value. Seems like the least janky option to me.

Reply:
I think it will be confusing if you suddenly switch to the left or right tab. I would disable switching between tabs while the input component is in focus.

Reply:
Yeah, I agree, but I guess there's a potential scenario where you end up with nothing but an input field and you need a way of escaping. I've pushed a change now that disables it by default, but allows you to enable tab switching with a prop.